consensus/polybft/consensus_runtime.go

```diff
@@ -1114,9 +1114,17 @@ func (c *consensusRuntime) HasQuorum(
 	}
 
 	// extract the addresses of all the signers of the messages
-	signers := make([]types.Address, len(messages))
-	for i, message := range messages {
-		signers[i] = types.BytesToAddress(message.From)
+	signers := make(map[types.Address]struct{}, len(messages))
+
+	for _, message := range messages {
+		if message.Type != msgType {
+			c.logger.Warn("HasQuorum received not matching message and message type",
+				"type of message", message.Type, "submitted message type", msgType)
+
+			return false
+		}
```

Comment:
Not sure is it safe to add this check? Because I have some flashback that @vcastellm told me that messages could be mixed up (at least it was the case for pbft consensus, where we had `prepare` and `commit` messages mixed up in the same queue during `prepare` state).

Reply:
Yes mixed messages here https://github.com/0xPolygon/go-ibft/blob/main/core/ibft.go#L1048-L1051